src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -478,13 +487,17 @@ public void updateUsedElementWithPotentialFieldNameExpr(NameExpr expr) {
       String classFullName = exprDecl.asField().declaringType().getQualifiedName();
       updateUsedClassWithQualifiedClassName(classFullName);
       usedMembers.add(classFullName + "#" + expr.getNameAsString());
+      updateUsedClassWithPossibleTypeArgs(exprDecl.getType());
     }
   }
 
   /**
    * Updates the list of used classes with the given qualified class name and its corresponding
    * enclosing classes, if applicable (in the case of nested classes).
    *
+   * <p>If we are updating the type of a used member, we should call this method together with
+   * {@link TargetMethodFinderVisitor#updateUsedClassWithPossibleTypeArgs(ResolvedType)}.
```

Comment:
"together with" is imprecise, because there is a required (or at least expected) ordering: first call this method, then call `updateUsedClassWithPossibleTypeArgs`.

Please also provide a precise definition of `used member` here so that it is clear to a reader what you mean.

Reply:
There is no ordering required, since these two methods do not rely on each other to work. 

> Please also provide a precise definition of used member here so that it is clear to a reader what you mean.


Given that we've already have a field called `usedMembers`, and the Javadoc of that field has already defined what "members" means, I think we can skip the definition of used member in the Javadoc of this method.

Reply:
> Given that we've already have a field called usedMembers, and the Javadoc of that field has already defined what "members" means, I think we can skip the definition of used member in the Javadoc of this method.

The definition of "used member" can just be "(i.e., a member of the `usedMembers` field of this class)". The point is just to make sure that a reader knows what you are talking about (and this is a big class with many interacting parts, so it's easy for a non-expert to get lost). Put another way, it's better to make these kind of links explicit rather than expecting the reader to remember that the class has a field with the appropriate name. (Also, we might change the name of the field later, and then this documentation might become out of date without anyone noticing.)

> There is no ordering required, since these two methods do not rely on each other to work.

Fair enough. We can skip this part.